front/src/applications/stdcm/utils/checkStdcmConfigErrors.ts

```diff
@@ -28,86 +30,142 @@ const checkStdcmConfigErrors = ({
+
+  if (origin.location && destination.location) {
+    if (
+      origin.location.uic === destination.location.uic &&
+      origin.location.secondary_code === destination.location.secondary_code
+    ) {
+      routeErrors.push(StdcmConfigErrorTypes.ZERO_LENGTH_PATH);
     }
-    if (!destination.location) {
-      missingFields.push('destination');
+
+    const isOriginRespectDestinationSchedule =
+      !origin.isVia && origin.arrivalType === ArrivalTimeTypes.RESPECT_DESTINATION_SCHEDULE;
+    const isDestinationASAP =
+      !destination.isVia && destination.arrivalType === ArrivalTimeTypes.ASAP;
+
+    if (isOriginRespectDestinationSchedule && isDestinationASAP) {
+      routeErrors.push(StdcmConfigErrorTypes.NO_SCHEDULED_POINT);
     }
 
-    if (missingFields.length > 0) {
-      return {
-        errorType: StdcmConfigErrorTypes.MISSING_INFORMATIONS,
-        errorDetails: { missingFields },
-      };
+    const areBothPointsScheduled =
+      !origin.isVia &&
+      !destination.isVia &&
+      origin.arrivalType === ArrivalTimeTypes.PRECISE_TIME &&
+      destination.arrivalType === ArrivalTimeTypes.PRECISE_TIME;
+
+    if (areBothPointsScheduled) {
+      routeErrors.push(StdcmConfigErrorTypes.BOTH_POINT_SCHEDULED);
+      routeErrorDetails.originTime = origin.arrival
+        ? t('leaveAt', { time: dateToHHMMSS(origin.arrival, { withoutSeconds: true }) })
+        : t('departureTime');
+      routeErrorDetails.destinationTime = destination.arrival
+        ? t('arriveAt', { time: dateToHHMMSS(destination.arrival, { withoutSeconds: true }) })
+        : t('destinationTime');
     }
   }
 
+  if (pathfindingStatus === 'failure') {
+    routeErrors.push(StdcmConfigErrorTypes.PATHFINDING_FAILED);
+  }
+
   if (origin.isVia) {
     throw new Error('First step can not be a via');
   }
   if (destination.isVia) {
     throw new Error('Last step can not be a via');
   }
 
-  if (
-    origin.location?.uic === destination.location?.uic &&
-    origin.location?.secondary_code === destination.location?.secondary_code
-  ) {
-    return { errorType: StdcmConfigErrorTypes.ZERO_LENGTH_PATH };
-  }
+  stdcmPathSteps.forEach((step) => {
+    if (step.isVia) {
+      const { stopType, stopFor } = step;
 
-  if (pathfindingStatus && pathfindingStatus === 'failure') {
-    return { errorType: StdcmConfigErrorTypes.PATHFINDING_FAILED };
-  }
-
-  const isOriginRespectDestinationSchedule =
-    origin.arrivalType === ArrivalTimeTypes.RESPECT_DESTINATION_SCHEDULE;
+      if (
+        (stopType === 'driverSwitch' || stopType === 'serviceStop') &&
+        (stopFor === undefined || stopFor === null)
+      ) {
+        routeErrors.push(StdcmConfigErrorTypes.VIA_STOP_DURATION_MISSING);
+      }
 
-  const isDestinationASAP = destination.arrivalType === ArrivalTimeTypes.ASAP;
+      if (stopType === 'driverSwitch' && stopFor !== undefined && stopFor !== null && stopFor < 3) {
```

Comment:
Same at l.124 (for duration issue)
```suggestion
      if (stopType === 'driverSwitch' && !isNil(stopFor) && stopFor.total('minute') < 3) {
```

Reply:
Better this way !  Changed